worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala

```diff
@@ -1246,6 +1215,28 @@ class PushDataHandler(val workerSource: WorkerSource) extends BaseMessageHandler
     }
   }
 
+  def writeDataWithExceptionHandling(
+      fileWriter: FileWriter,
+      body: ByteBuf,
+      shuffleKey: String): Future[Unit] = {
+    Future {
+      try {
+        fileWriter.write(body)
+      } catch {
+        case e: AlreadyClosedException =>
+          fileWriter.decrementPendingWrites()
+          val (mapId, attemptId) = getMapAttempt(body)
+          val endedAttempt =
+            if (shuffleMapperAttempts.containsKey(shuffleKey)) {
+              shuffleMapperAttempts.get(shuffleKey).get(mapId)
+            } else -1
+          logWarning(s"Append data failed for task(shuffle $shuffleKey, map $mapId, attempt" +
```

Comment:
should this be error logging?

Reply:
yes, that would be a good change. Let me make that change.